webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/CreateNodeData.scala

```diff
@@ -85,13 +82,20 @@ object Rest {
 
   final case class NodeProp(name: String, value: ConfigValue)
 
+  // machine creation doesn't have an ID
+  final case class MachineCreate(`type`: String, provider: Option[String], manufacturer: Option[String])
+
+  /*
+   * We can have two kinds of node where machine or machine type is given
+   */
   final case class NodeDetails(
       id:             String,
       hostname:       String,
       status:         String,
       os:             OS,
       policyServerId: Option[String],
-      machineType:    String,
+      machineType:    Option[String],
+      machine:        Option[MachineCreate],
```

Comment:
it looks like we relax a bit things by making both optional, but in fact, we were already using `physical` as a default value.